language/stdlib/modules/LibraAccount.move

```diff
@@ -1102,6 +1139,29 @@ module LibraAccount {
         )
     }
 
+    /// The prologue for WriteSet transaction
+    fun writeset_prologue(
+        sender: &signer,
+        writeset_sequence_number: u64,
+        writeset_public_key: vector<u8>,
+        txn_expiration_time: u64,
+        chain_id: u8,
+    ) acquires LibraAccount, Balance {
+        assert(Roles::has_libra_root_role(sender), Errors::invalid_argument(PROLOGUE_INVALID_WRITESET_SENDER));
```

Comment:
The previous version of this code had an assert for "sender == CoreAddresses::LIBRA_ROOT_ADDRESS()", as well as the has_libra_root_role assert here. Did you remove that intentionally?

Reply:
Yes this is intentional. IIUC `sender == CoreAddresses::LIBRA_ROOT_ADDRESS()` is a hard coded version of `has_libra_root_role`. Am I doing this right @tzakian?

Reply:
has_libra_root_role does not have an address-specific check (only assert_libra_root) IIRC. So I would add the address check here as well.